.ado/pipelines/templates/jobs-workload-deploy.yaml

```diff
@@ -20,10 +20,41 @@ jobs: # using jobs so they each run in parallel
         scriptLocation: inlineScript
         inlineScript: |
 
-          helm lint src/app/charts/healthservice # lint helm chart
+          # Must match the workload name in the workloads JSON file
+          $workloadName = "healthservice"
 
           # load json data from downloaded terraform artifact
           $releaseUnitInfraDeployOutput = Get-ChildItem $(Pipeline.Workspace)/terraformOutputReleaseUnitInfra/*.json | Get-Content | ConvertFrom-JSON
+          $globalInfraDeployOutput = Get-ChildItem $(Pipeline.Workspace)/terraformOutputGlobalInfra/*.json | Get-Content | ConvertFrom-JSON
+
+          $acrName = $globalInfraDeployOutput.acr_name.value # Container Registry Name (from terraform output)
+          $acrUrl = $globalInfraDeployOutput.acr_login_server.value # Container Registry Login Server (from terraform output)
+
+          $workloadsFile = "$(System.DefaultWorkingDirectory)/.ado/pipelines/config/workloads.json"
+
+          if(-not (Test-Path $workloadsFile))
+          {
+            throw "*** Container workload file not found at $workloadsFile"
+          }
+
+          $workloads = Get-Content $workloadsFile | ConvertFrom-JSON
+          $workload = $workloads.$workloadName
+
+          if(-not $workload)
+          {
+            throw "*** Workload '$workloadName' not found in $workloadsFile"
+          }
+
+          $chartRegistry,$chartName = $workload.chart.name -split "/",2
+
+          # if the helm chart is hosted on ACR, we need to log in first
+          if($chartRegistry.EndsWith(".azurecr.io"))
+          {
+            az acr login -n $chartRegistry
```

Comment:
This is something I've though about in #293 but not implemented it yet. You're assuming that service connection has access to the source registry, right?

Reply:
yep, I would start with that. Otherwise it will get messy